test/unit/test_host.rb

```diff
@@ -65,8 +65,14 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com').eql? Host.new('example.com')
-      assert Host.new('example.com').equal? Host.new('example.com')
+      h1 = Host.new('first_host')
+      h2 = Host.new('second_host')
+      h1.expects(:hash).times(3).returns('example.com')
```

Comment:
I like extracting the h1 and h2 variables for the hosts, but I think the expectation lines are quite problematic for a couple of reasons:

1) IMO it is misleading that first_host and second_host would have the same hash, they should have different hashes because the hosts are different. The test sets up the opposite of what would happen in real life.
2) hash returns a number not a string - returning 'example.com' is misleading

I wonder if the following would appease rubocop:

```ruby

def test_assert_hosts_compare_equal
  h1 = Host.new('example.com')
  h2 = Host.new('example.com')

  assert h1 == h2
  assert h1.eql? h2
  assert h1.equal? h2
end

```